contracts/utils/Strings.sol

```diff
@@ -64,4 +64,36 @@ library Strings {
         require(value == 0, "Strings: hex length insufficient");
         return string(buffer);
     }
+
+    /**
+     * @dev Converts each character ('a' to 'z') of a `string` to it's upper-case form ('A' to 'Z').
+     * Requires parameter string `value` to be comprised purely of basic-ASCII characters.
+     */
+    function normalizeUpperASCII(string calldata value) internal pure returns (string memory) {
+        bytes memory buffer = bytes(value);
+        for (uint256 i = 0; i < buffer.length; i++) {
+            require(buffer[i] & 0x80 == 0, "Basic ASCII string required."); // 0x80-bit on is non-ASCII.
+            if (buffer[i] >= "a" && buffer[i] <= "z") { // only work on lower-case characters.
+                buffer[i] &= 0x1F; // mask-off the lower-case bit.
```

Comment:
This is not good. What you want to do is turn of the 6th bit (that is work 32)
Is is done by doing `^= 0x20`

`&= 0x1F` is a completely different operation that makes no sens here.

Reply:
^= toggles bits. If somehow an uppercase later made it through the separate lines of code above, it would produce a lowercase character instead of uppercase.

&= allows us to take all the bits, *except* the casing bit. Though I do agree &= 0x1F is incorrect, as that also relies on the `require()` above it to exist. I think `&= 0xDF` would be more reasonable.

Reply:
How do you expect an uppercase letter to make it through here? Under what circumstances would an uppercase pass `buffer[i] >= "a" && buffer[i] <= "z"` ???

Reply:
`&= 0xDF` sounds good !

Reply:
> How do you expect an uppercase letter to make it through here? Under what circumstances would an uppercase pass `buffer[i] >= "a" && buffer[i] <= "z"` ???

Under possible future circumstances where that line of code becomes bugged.

Reply:
> `&= 0xDF` sounds good !

It's now been changed to `&= ~bytes1(0x20)` (which equates to 0xDF).